nicosrm/24-emb-chess
# Add general chess piece structure, chess pieces and move validation.

# Changes
- Add general structure for chess pieces, i.e. abstract `ECPChessPiece` class
- Add helper classes (`ECPChessField`, `ECPBoardColumn`)
- Add pawn, rook, king, knight, and queen incl. test sketch for `isValid` method. #4 #9

# Note
- Note that the `move` method is not (fully) implemented, yet. The actual movement of the Dezibot on the board is missing for example. This will be done in #4 as well.

# How to Test
- Since this is only about the structure, compiling the example sketch should be sufficient.
- Check `src/chess_piece_move_validation_tests` sketches to verify `ECPPawn`, `ECPRook`, `ECPKing`, and `ECPKnight` implementation.

Closes #20 #9 #10 #4
